v2/logging.go

```diff
@@ -1,23 +1,50 @@
 package shuttle
 
 import (
-	"io"
+	"context"
 	"log/slog"
 )
 
-// logging is disabled by default
-var log = slog.New(slog.NewTextHandler(io.Discard, nil))
+type Logger interface {
+	Info(s string)
+	Warn(s string)
+	Error(s string)
+}
+
+var getLogger = func(ctx context.Context) Logger {
+	return &contextLogger{ctx: ctx, logger: slog.Default()}
+}
 
-func getLogger() *slog.Logger {
-	return log
+// Deprecated: Use SetLogHandler instead to adapt slog.
+// SetLoggerFunc sets the function to be used to acquire a logger when go-shuttle logs.
+func SetLoggerFunc(fn func(ctx context.Context) Logger) {
+	getLogger = fn
 }
 
 // SetLogHandler allows to set a custom slog.Handler to be used by the go-shuttle logger.
-// If handler is nil, the default slog logger will be used.
+// If handler is nil, the default slog handler will be used.
 func SetLogHandler(handler slog.Handler) {
 	if handler == nil {
-		log = slog.Default()
-	} else {
-		log = slog.New(handler)
+		handler = slog.Default().Handler()
 	}
+	getLogger = func(ctx context.Context) Logger {
+		return &contextLogger{ctx: ctx, logger: slog.New(handler)}
+	}
+}
+
+type contextLogger struct {
+	ctx    context.Context
```

Comment:
add a comment to acknowledge that storing the ctx is bad practice, and tha it will go away when we finish deprecating the previous log configuration